### DISCUSSION TITLE
Getting syntax errors on Windows with VS2017 ...

### DISCUSSION THREAD

**Discussion by shock-pro** [ginkgo-project/ginkgo#825] [Q&A]
Hi Everyone,
I am using the released version ginkgo-1.3.0,  CMAKE 3.20 finished fine, only 3 warnings for developers, nothing else. 
Then exported the VS solutions fine, but then VS is having many syntax error as shown below:
![image](https://user-images.githubusercontent.com/77303301/124659113-36aae080-de6a-11eb-9d30-abf6a96b9822.png)

![image](https://user-images.githubusercontent.com/77303301/124659199-4f1afb00-de6a-11eb-86ce-b077ed27e103.png)

Please, let me know how to fix this issue.
Thanks and Regards,

**Comment by yhmtsai** [comment]
Hi @shock-pro, thanks for reporting this issue.
we faced it in some tests before.
Could you try change `{...}` to `(...)`?
Could you also provide the VS2017 version?

**Reply by shock-pro** [reply]
Hi Yhmtsai!
That change fixed the issue. I don't think that curly braces are allowed in that context, are they ?
I am using:
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community>cl
Microsoft (R) C/C++ Optimizing Compiler Version 19.16.27045 for x64

After fixing the curly braces, I am getting hundreds of errors:
For example:
![image](https://user-images.githubusercontent.com/77303301/124664241-c6538d80-de70-11eb-92da-d435da2b294b.png)

![image](https://user-images.githubusercontent.com/77303301/124664272-ceabc880-de70-11eb-9240-3cb4959e90f9.png)

Do you use VS to build Ginkgo often ?
Thanks!

**Reply by yhmtsai** [reply]
The error number is a little weird. Does it become C2397 on their website?
If it is list initialization such that it can not be narrowed conversion, it is weird gcc/clang pass them.
We have Github Action to use VS2019 to test compilation on reference/CUDA and running/compilation on reference.

**Reply by shock-pro** [reply]
Well, there are also hundreds of issues all related to the unsigned size_t used for gko::size_type

![image](https://user-images.githubusercontent.com/77303301/124670930-4f22f700-de7a-11eb-8ac5-ba9182b04e55.png)

![image](https://user-images.githubusercontent.com/77303301/124670954-5813c880-de7a-11eb-9a4a-75d0c004ed94.png)

If you can change gko::size_type to a signed int, that would all the OMP for loops.
But I don't know if this may cause problems in other parts of the code.
Cheers,

**Reply by shock-pro** [reply]
I tried changing the definition of gko::size_type but as expected new incompatibility type errors appear in other places:
![image](https://user-images.githubusercontent.com/77303301/124672647-03be1800-de7d-11eb-959f-01ee7d204ade.png)

**Reply by yhmtsai** [reply]
please ignore the OMP error. We do not support OMP in MSVC.
you can use cmake -DGINKGO_BUILD_OMP=OFF .. to disable it.
the current develop branch disabling it automatically, but 1.3.0 still needs to turn off manually

**Comment by shock-pro** [comment]
If we modify gko::size_type to fix the C3016 issue, we get new issues in other places:
![image](https://user-images.githubusercontent.com/77303301/124684350-86060680-de94-11eb-93c3-d41a2bb344a7.png)
